# Fix the makefile to use EB values for everything
# Based off magma-2.3.0-fix-makefile.patch by wpoely86@gmail.com
# and updated for 2.5.0
--- magma-2.5.0/Makefile.orig	2019-03-08 12:33:11.872321116 +0000
+++ magma-2.5.0/Makefile	2019-03-08 12:35:58.463432082 +0000
@@ -10,7 +10,7 @@
 CC         ?= cc
 CXX        ?= c++
 NVCC       ?= nvcc
-FORT       ?=
+FORT       ?= $(FC)
 ifeq ($(FORT),)
     $(warning No Fortran compiler was given in FORT in make.inc. Some testers will not be able to check their results.)
 endif
@@ -23,17 +23,17 @@
 #FPIC       = -fPIC
 
 # may want -std=c99 for CFLAGS, -std=c++11 for CXXFLAGS
-CFLAGS     ?= -O3 $(FPIC) -DADD_ -Wall -MMD
-CXXFLAGS   ?= $(CFLAGS) -std=c++11
-NVCCFLAGS  ?= -O3         -DADD_ -Xcompiler "$(FPIC) -Wall -Wno-unused-function" -std=c++11
-FFLAGS     ?= -O3 $(FPIC) -DADD_ -Wall -Wno-unused-dummy-argument
-F90FLAGS   ?= -O3 $(FPIC) -DADD_ -Wall -Wno-unused-dummy-argument
-LDFLAGS    ?= -O3 $(FPIC)
+CFLAGS     := $(FPIC) -DADD_ -Wall -MMD $(CFLAGS)
+CXXFLAGS   := $(CFLAGS)
+NVCCFLAGS  :=         -DADD_ -Xcompiler "$(FPIC) -Wall -Wno-unused-function $(CFLAGS)"
+FFLAGS     := $(FPIC) -DADD_ -Wall -Wno-unused-dummy-argument $(FCFLAGS)
+F90FLAGS   := $(FPIC) -DADD_ -Wall -Wno-unused-dummy-argument $(F90FLAGS)
+LDFLAGS    := $(FPIC) -fopenmp $(LDFLAGS)
 
 INC        ?= -I$(CUDADIR)/include
 
 LIBDIR     ?= -L$(CUDADIR)/lib64
-LIB        ?= -lcudart -lcudadevrt -lcublas -lcusparse -llapack -lblas -lpthread -lm
+LIB        := $(LIBLAPACK) -lcudart -lcudadevrt -lcublas -lcusparse -lpthread -lm $(LIBS)
 
 GPU_TARGET ?= Kepler Maxwell Pascal
 
